src/main/java/taxiservice/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,238 @@
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            } catch (SQLException e) {
+                throw new DataProcessingException("Can't create " + car, e);
+            }
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        Car result = null;
+        String query = "SELECT * FROM " + CARS_TAB_NAME + " c"
+                + " INNER JOIN " + MANUFACTURERS_TAB_NAME + " m"
+                + " ON c.manufacturer_id = m.manufacturer_id"
+                + " WHERE car_id = ? AND c.deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                result = newCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get by id " + id, e);
+        }
+
+        if (result != null) {
+            result.setDrivers(getDrivers(result.getId()));
+        }
+        return Optional.ofNullable(result);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        List<Car> result = new ArrayList<>();
+        String query = "SELECT * FROM " + CARS_TAB_NAME + " c"
+                + " INNER JOIN " + MANUFACTURERS_TAB_NAME + " m"
+                + " ON c.manufacturer_id = m.manufacturer_id"
+                + " WHERE c.deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                result.add(newCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all", e);
+        }
+        if (result != null) {
+            for (Car car : result) {
+                car.setDrivers(getDrivers(car.getId()));
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE " + CARS_TAB_NAME
+                + " SET model = ?,"
+                + " manufacturer_id = ? "
+                + "WHERE car_id = ? AND deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update " + car, e);
+        }
+
+        query = "DELETE FROM " + CARS_DRIVERS_TAB_NAME
```

Comment:
create separate method for delete and use same connection, create just new PreparedStatement 
```suggestion
        query = "DELETE FROM " + CARS_DRIVERS_TAB_NAME
```